File: Type of array/Type_of_array.cpp
class Solution {
  public:
    int maxNtype(vector<int>& arr) {
        int inc =0,dec = 0;
        int n = arr.size();
        for(int i =1; i<n; i++) {
            if(arr[i-1]<=arr[i]) inc++;
            else dec++;
        }
        if(inc == 0) return 2;
        else if(dec == 0) return 1;
        else if(dec > inc) return 3;
        return 4;
    }
};

File: Type of array/Type_of_array.py
class Solution:
    def maxNtype(self , arr):
        inc = 0
        dec = 0
        n = len(arr)

        for i in range(1, n):
            if arr[i - 1] <= arr[i]:
                inc += 1
            else:
                dec += 1

        if inc == 0:
            return 2
        elif dec == 0:
            return 1
        elif dec > inc:
            return 3
        return 4
